README.md

```diff
@@ -66,6 +66,15 @@ And dryrun
     assert changes == {"value": {'current': 'read', 'desired': 'write'}}
 ```
 
+Testing functions and classes defined in a module is also possible 
+by simply importing them after the project is compiled
+
+```python
+project.compile("") # Model to compile
```

Comment:
Compiling an empty model looks quite strange and can be quite confusing. In my opinion the functions and classes should be loaded automatically, as such that the compile step is not required anymore. @wouterdb wdyt?

Reply:
It is a bit of a tricky situation.

1. The module loader that is now in the core allows loading plugins from modules that are not actually imported (but just happen to be there on disk) (this is why this works without an import in the main file)
2. By simply enabling the module loader, the import will just work, unless there are dependencies on other modules that are required by this one, but that haven't been downloaded.
3. The get_plugin_function call could now also get this problem, that plugins can fail to load due to dependencies on other modules. 

For this case, I would opt for the solution that @andraskvr has implemented now, but I would not use an empty main file, but one that imports the module under test. 

For the get_plugin_function, to work on the same principle would be a huge performance regression. 

So there is a case to be made to do the load when the global fixture is set up, from that point on, the modules can always be loaded.  It will change the way pytest fails when the project fails to load, but not too much.